### DISCUSSION TITLE
new ideas

### DISCUSSION THREAD

**Discussion by Amin456789** [LostRuins/koboldcpp#875] [Ideas]
hi
im back with  new ideas

1, sd ui = when we choose a random seed and choose a batch count like 3 it will make 3 same images of same seed. please make it to add +1 to the seed while we doing randomizer

2. please add seed information to random seeds that use -1 cause it wont give us the seed in sd ui

3. something i found that the lcm model i use sometimes 512x512 images are not being made in sd ui and give us black image [sdxl turbo works fine] but other resoltuons works great like 512x448 or whatver else. maybe in chat when we have custom resolution we can fix the black images for some of the lcm models too

4.automatic image geneation in chat is great. but wish we could select which part of chats can be generated without typing it

5. it seems the images are being generated in chat are somehow low quality, even we choose square, please make it the same as oriiginal resolution and put a download button for it

6.img2img batch count  for sd ui seems not working, please fix it if possible, leejet knows myabe how

thank u for ur hard work

**Comment by LostRuins** [comment]
For fixing black images, use the fix vae option or `--sdvaeauto`

For higher res, ensure the "Save High Res" option is selected
![image](https://github.com/LostRuins/koboldcpp/assets/39025047/eb4e0464-f874-413c-ba73-67db6623dd4d)

**Reply by Amin456789** [reply]
thank u, i will use save higher res now!
i used fix sd vae too but still a bit problem for just this model, but its ok i use sdxl turbo for chat

**Comment by Amin456789** [comment]
7. controlnet for sd ui = sd cpp hast a very good working openpose controlenet, it will be great to have and as a bonus having one of those a1111 extensions for drawing poses will be great, there is a 3d one and we can add 2 more characters i think
8.  clip skip support for sd ui = sd cpp hast it for finding settings for default but sometimes its good we choose it ourselves

**Reply by LostRuins** [reply]
Clip skip support is already added. SD UI should be able to use it.

**Reply by Amin456789** [reply]
thanku

**Comment by Amin456789** [comment]
9. chat and chat ui are saving everything in browser which is great, wish sd ui can save the setting automaticly like chat, for example when we choose this sample or cfg in sd ui and when we open chrome we dont need to do it again

**Comment by Amin456789** [comment]
11. when we save the chats or anything it will save all the images automaticly so when we load them they will load those images as well [from the folder the exported] so when we delete cache in chrome they wont be deleted forever
12. we can move images in chats and anything else in the chat or we can generate a new image from earlier lines so it wont make the image in the end only

sorry for all of these suggestions, i just love this app and it will be great we will have them someday, i just try to make new suggestions for better experince for my usage which i beleve it will be good for others and hope for others to like it

**Reply by LostRuins** [reply]
Regarding 11, you can export the save to a json file on your PC which will include the images inside too.

For 12, you can easily enter edit mode and move images around within the story.

**Reply by Amin456789** [reply]
thank u for the answer, it is interssting that the pics will be included in a  single json file, didnt knew it

**Comment by Amin456789** [comment]
13. new leejet's sd cpp has a new update which makes the model loading much much faster. please consider updating to the last one
14. also supporting photomaker will be great too so we can use it with sdxl models in ssd ui. sd cpp supports it

**Comment by Amin456789** [comment]
14. auto update = having auto update will be great without redownloading the exe ourselves 
15.  making the app to use its presets with a narrow down button = it will be great to just choose presets and models with making the app to auto detect them with a narrow down featre. like we just click on them and it auto save the settings in the same folder as the exe upon loading the app so this way we wont be needing to make presets for every model

i made a topic in sd cpp so tell people how amazing ur app is, and hoping to help u with the sd cpp features to bring them here, thank u for ur hard work
